packages/framework/block-std/src/edgeless/bound.ts

```diff
@@ -0,0 +1,298 @@
+import {
+  EPSILON,
+  getBoundsFromPoints,
+  lineIntersects,
+  polygonPointDistance,
+} from '../utils/math.js';
+import { deserializeXYWH, serializeXYWH } from '../utils/xywh.js';
+import type { IBound, SerializedXYWH, XYWH } from './types.js';
+import type { IVec } from './vec.js';
+
+export class Bound implements IBound {
```

Comment:
Do you want to use `signal` to build this? properties like points seems can be auto computed.

Reply:
I would like to, but not in this PR.